E-Commerce/readme.md

```diff
@@ -46,13 +46,101 @@ erDiagram
 Write SQL queries to complete the following tasks:
 
 - [ ] List all the products whose name contains the word "socks"
+
+SELECT * 
+FROM products
+WHERE product_name LIKE '%socks%';
+
+
+
 - [ ] List all the products which cost more than 100 showing product id, name, unit price, and supplier id
+
+ SELECT p.id, p.product_name, pa.unit_price, pa.supp_id
+ FROM products AS p
+ INNER JOIN product_availability AS pa
+ ON p.id = pa.prod_id
+ WHERE pa.unit_price > 100;
+
+
+
+
 - [ ] List the 5 most expensive products
+
+SELECT p.id, p.product_name, pa.unit_price
+ FROM products AS p
+ INNER JOIN product_availability AS pa
+ ON p.id = pa.prod_id
+ ORDER BY pa.unit_price DESC 
+ LIMIT 5;
+
+
+
+
 - [ ] List all the products sold by suppliers based in the United Kingdom. The result should only contain the columns product_name and supplier_name
+
+SELECT p.product_name, s.supplier_name
+FROM products AS p
+INNER JOIN product_availability AS pa
+ON p.id = pa.prod_id
+INNER JOIN suppliers AS s
+ON pa.supp_id = s.id
+WHERE s.country = 'United Kingdom';
+
+
+
 - [ ] List all orders, including order items, from customer named Hope Crosby
+
+SELECT  
+    oi.product_id,
+    oi.order_id,
+    o.order_date,
+    o.order_reference,
+    o.customer_id,
+    c.name AS customer_name
+FROM orders AS o
+INNER JOIN order_items AS oi
+ON o.id = oi.order_id
+INNER JOIN customers AS c 
+ON o.customer_id = c.id
+WHERE c.name = 'Hope Crosby';
+
+
+
 - [ ] List all the products in the order ORD006. The result should only contain the columns product_name, unit_price, and quantity
+
```

Comment:
I see the same product multiple times in the result list. Have another look at the join with `product_availability`. How do we figure out which row in `product_availability` matches an order item? Does it only depend on the product id, or is there another important column involved?